src/java/main/org/apache/zookeeper/server/quorum/QuorumZooKeeperServer.java

```diff
@@ -39,6 +41,7 @@
 
     public final QuorumPeer self;
     protected UpgradeableSessionTracker upgradeableSessionTracker;
+    private final ZooKeeperServerListener listener = new QuorumListenerImpl(this);
```

Comment:
The parent class also has a private member that is a listener. Do we really need two listeners for a QuorumZooKeeperServer instance?